publication/src/main/scala/hmda/publication/reports/util/ReportUtil.scala

```diff
@@ -62,6 +69,52 @@ object ReportUtil extends SourceUtils {
     )
   }
 
+  def applicantIncomesWithBorrowerCharacteristics[ec: EC, mat: MAT, as: AS](
```

Comment:
It was hard to find a meaningful name for this method, since the data structure it produces is nested so deeply. I'm open to suggestions for improved names.

Reply:
We seem to be following a pattern of `calculateThis` with the methods we're calling.  Maybe `calculateApplicantIncomes`?  It returns a list of Applicant Income objects, and it's assigned to a value called `applicantIncomesF`.

Reply:
Hmm. I hadn't thought about the inconsistency of the naming conventions in the `ReportUtil` file. The question of when to name a method with a verb (like `calculateDispositions`) vs with a noun (like `larsByIncomeInterval`) is interesting--I'm still forming my opinion on it. 

In our system, these methods don't have side effects, so the only part that matters is the return value. Thus, our code can kind of treat them like objects, so I think I prefer naming them as nouns. If we wanted to make them all consistent, I would vote to never use a verb, and to switch them all to be named with nouns. 

This kind of consistency isn't 100% essential, but I don't think I want to add any functions with verb names.

Reply:
I could use `applicantIncomesByInterval` if we want a shorter name?